### ISSUE TITLE
Convert services to daemons in snap

### ISSUE THREAD

**Issue by anonymouse64** [edgexfoundry/edgex-go#485]
We want to convert all of the services to daemons so that the snap is able to start up on boot automatically and systemd can manage services going down and coming back up, etc. which currently the snap does not do at all.

**Comment by anonymouse64** [comment]
Currently, due to a bug in snapd (see [launchpad bug # 1796125](https://bugs.launchpad.net/snapd/+bug/1796125)), service ordering is not respected on installation of a snap. This affects the security services, most of which such as kong, the vault-worker, etc. are not tolerant of having their dependency services not started up when they start up and so for Delhi, we will manage the dependencies manually by putting all of the security services into a single, forking daemon.

Additionally, due to another bug in snapd, services can't be disabled from the install hook, and as such when the snap is installed all services startup immediately, whether they were disabled in the install hook or not. A fix for this is in flight and should land for snapd 2.36, but since in the meantime all of the EdgeX services (with the exception of security services) are able to wait for their dependencies when they start up, it's okay to have all of the services startup on install.
Once the fix lands in 2.36, then folks installing the snap will only have the minimal set of services startup as intended.

I will file a separate issue post-Delhi to handle the security services being split up into individual services.

**Comment by tonyespy** [comment]
This has landed in Dehli (#722), but is not yet in master, so adjusting labels accordingly.
